addons/hr_expense/hr_expense.py

```diff
@@ -1,324 +1,272 @@
-        'date': fields.date.context_today,
-        'state': 'draft',
-        'employee_id': _employee_get,
-        'user_id': lambda cr, uid, id, c={}: id,
-        'currency_id': _get_currency,
-    }
-
-    def unlink(self, cr, uid, ids, context=None):
-        for rec in self.browse(cr, uid, ids, context=context):
-            if rec.state != 'draft':
+    name = fields.Char(readonly=True)
+    date = fields.Date(readonly=True, default=fields.Date.context_today)
+    journal_id = fields.Many2one('account.journal', string='Force Journal', readonly=True, states={'draft': [('readonly', False)], 'confirm':[('readonly', False)]}, help = "The journal used when the expense is done.")
+    employee_payable_account_id = fields.Many2one('account.account', string='Employee Account', readonly=True, states={'draft': [('readonly', False)], 'confirm':[('readonly', False)]}, help="Employee payable account")
+    employee_id = fields.Many2one('hr.employee', string="Employee", required=True, readonly=True)
+    date_confirm = fields.Date(string='Confirmation Date', copy=False, readonly=True,
+                                help="Date of the confirmation of the sheet expense. It's filled when the button Confirm is pressed.")
+    date_valid = fields.Date(string='Validation Date', copy=False, readonly=True,
+                              help="Date of the acceptation of the sheet expense. It's filled when the button Accept is pressed.")
+    user_valid = fields.Many2one('res.users', string='Validation By', readonly=True, copy=False,
+                                  states={'draft': [('readonly', False)], 'confirm':[('readonly', False)]})
+    account_move_id = fields.Many2one('account.move', string='Ledger Posting', copy=False, readonly=True, track_visibility="onchange")
+    line_ids = fields.One2many('hr_expense.expense', 'expense_id', string='Expense Lines', copy=True,
+                                readonly=True, states={'draft': [('readonly', False)]} )
+    note = fields.Text()
+    amount_untaxed = fields.Float(compute='_amount', string='Untaxed Amount', digits=dp.get_precision('Account'), store=True)
+    amount_tax = fields.Float(compute='_amount', string='Taxes', digits=dp.get_precision('Account'), store=True)
+    amount = fields.Float(compute='_amount', string='Total', digits=dp.get_precision('Account'), store=True)
+    currency_id = fields.Many2one('res.currency', string='Currency', required=True, readonly=True, states={'draft': [('readonly', False)], 'confirm': [('readonly', False)]}, default=lambda self:self.env.user.company_id.currency_id.id)
+    department_id = fields.Many2one('hr.department', string='Department', readonly=True, states={'draft': [('readonly', False)], 'confirm': [('readonly', False)]})
+    company_id = fields.Many2one('res.company', string='Company', required=True, readonly=True, states={'draft': [('readonly', False)], 'confirm':[('readonly', False)]}, default=lambda self: self.env.user.company_id)
```

Comment:
Wrong value for default !!!
Use:
default=lambda self: self.env['res.company']._company_default_get('hr.employee')